shared/dev/dumb-sheet.render.native.js

```diff
@@ -40,32 +37,33 @@ class Render extends Component<void, Props, any> {
     }, 0)
   }
 
-  _continueTest () {
-    if (isTesting && this.state.testIndex !== -1) {
+  _increment () {
+    if (this.props.autoIncrement && this.state.testIndex !== -1) {
       const {componentsOnly} = this._getComponents()
       if (this.state.testIndex >= componentsOnly.length) {
         this.setState({testIndex: -1})
       } else {
-        this.setState({testIndex: this.state.testIndex + TEST_RENDER_NUMBER})
+        this.setState({testIndex: this.state.testIndex + 1})
       }
     }
   }
 
   componentDidMount () {
-    if (isTesting) {
-      this._continueTest()
+    if (this.props.autoIncrement) {
+      this._increment()
     }
   }
 
   componentDidUpdate () {
-    setTimeout(() => {
-      this._continueTest()
-    }, 1)
+    if (this.props.autoIncrement) {
+      setTimeout(() => {
+        this._increment()
+      }, 1)
```

Comment:
Sure but again this component is for us to test our components so I'm not
looking for an elegant architecture for this thing

On Wednesday, August 31, 2016, Max Goodman <notifications@github.com> wrote:

> In shared/dev/dumb-sheet.render.native.js
> <https://github.com/keybase/client/pull/4053#discussion_r77088370>:
>
> >      }
> >    }
> >
> >    componentDidUpdate () {
> > -    setTimeout(() => {
> > -      this._continueTest()
> > -    }, 1)
> > +    if (this.props.autoIncrement) {
> > +      setTimeout(() => {
> > +        this._increment()
> > +      }, 1)
>
> Yeah, I understand. It's extremely hacky. This "cycling through the views"
> test behavior is not a mode the component should implement; components
> should only care about rendering themselves and responding to user
> interactions.
>
> FWIW the right way to do this would be to control the view using props and
> call ReactDOM.render, which provides a second argument callback once the
> rendering is completed. Putting all of this into lifecycle methods of a
> shared component (which gets used in other places) is not separating
> concerns properly.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/keybase/client/pull/4053/files/dc32ebab67e2b5011be080959d5dc7301771135f..a2530d5f598003cb6416f7a8cff641c1c3ab9514#r77088370>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAfhUz6baRY1-IypYp46nE_szFuIsA_Yks5qlgYkgaJpZM4Jtgrl>
> .
>